src/lib.rs

```diff
@@ -251,6 +262,54 @@ fn map_activepool_events(blk: &eth::Block, events: &mut contract::Events) {
         .collect());
 }
 
+// #[substreams::handlers::map]
+// fn get_eth_price(events: contract::Events, chainlink_prices: StoreGetBigDecimal /*uniswap_prices: StoreGetProto<Erc20Price>*/) -> Result<contract::EtherSent, substreams::errors::Error> {
+//     // let price = uniswap_prices
+//     // .get_last("UsdPriceByTokenSymbol:ETH")
+//     // .map_or(Some(BigDecimal::from(0)), |price| BigDecimal::from_str(&price.price_usd).ok());
+
+//     let eth_price = chainlink_prices.get_last("price_by_symbol:ETH:USD").map_or(BigDecimal::from(0), |price| price);
+//     // let eth_price = uniswap_prices
+//     //     .get_last("UsdPriceByTokenSymbol:ETH")
+//     //     .map(|price| BigDecimal::from_str(&price.price_usd).ok())
+//     //     .unwrap_or_else(|| Some(BigDecimal::from(0)))
+//     //     .unwrap_or(BigDecimal::from(0));
+
+
+//     substreams::log::println(format!("{:?}", eth_price));
+
+//     let all_sents: Vec<contract::ActivepoolEtherSentUsd> = events.activepool_ether_sents
+//     .into_iter()
+//     .map(|original| {
+
+//         let convert_amount = BigInt::from_str(&original.u_amount).unwrap_or(BigInt::from(0)).to_decimal(BigInt::from(18).to_u64());
+
+//         contract::ActivepoolEtherSentUsd {
+//             u_amount: original.u_amount.to_string(),
+//             usd_value: eth_price.clone().mul(convert_amount).to_string()
+//         }
+//     })
+//     .collect();
+
+//     Ok(contract::EtherSent { sents: all_sents })
+// }
+
+
+fn get_eth_price(chainlink_prices: StoreGetBigDecimal) -> BigDecimal{
+    let eth_price = chainlink_prices.get_last("price_by_symbol:ETH:USD").map_or(BigDecimal::from(0), |price| price);
+    substreams::log::println(format!("{:?}", eth_price));
+    eth_price
+}
+
+
+fn get_name() -> String {
+    let name = abi::activepool_contract::functions::Name {};
+    let name_options = name.call(ACTIVEPOOL_TRACKED_CONTRACT.to_vec());
+    let x = name_options.unwrap();
```

Comment:
This will panic and error if `name_options` is `None`. Why don't you return an Option from the `get_name` function, and then determine what to do based on the return value where it is called. That way this function will never panic, and you handle each case in the calling code.